### PR TITLE
M3-5501: [BUILD] Update the changelog generation script

### PR DESCRIPTION
**Pull request by CodyFinn-Akamai**
## Description 📝

Updates the `generate_changelog.py` script in the following ways:
- Allows the user to specify what sub-repo the changelog should be generated for
- Adds the new naming conventions form PR name standardization proposal 
- Automatically strips Jira labels from changelog
- Automatically inserts current date

the syntax for calling the script has changed it should now take this form
`python generate_changelog.py <version number> <origin> <repo to generate changelog for>`
The repo argument is optional and only needed to specify `api-v4` or `validation`

## How to test 🧪

Ensure you can generate a changelog for the repos without error

### CODE REVIEW

**Review by dwiley-akamai — COMMENTED** [review]
This is looking solid so far, but I am seeing the release header format in the changelog (e.g., `## [2022-09-19] - v1.76.0`) is different now. Across the packages' changelogs, the format now has the version first and then the date -- e.g., `## [1.79.0] - 2022-10-19`. We should keep it consistent with what it has been historically

File: scripts/generate_changelog.py [hunk 1]
```diff
@@ -93,12 +101,19 @@ def generateChangeLog(release, date, origin):
...
+    else:
+        changelog_file = "packages/{}/CHANGELOG.md".format(repo)
+
+    read_change_log=open(changelog_file, 'r')
     change_log_lines=read_change_log.readlines()
     read_change_log.close()
 
     change_log_lines.insert(START_INSERT,'\n')
-    change_log_lines.insert(incrementLine(),'## [%s] - %s\n'%(release,date))
+    change_log_lines.insert(incrementLine(),'## [%s] - %s\n'%(date, release))
```
Comment (dwiley-akamai) [inline scripts/generate_changelog.py:116]: We should either add the `v` prefix here and note in the documentation that we don't have to pass it in, or vice versa, don't add it here but note in the documentation that we'll have to pass the `v` prefix with the version number

**Review by dwiley-akamai — COMMENTED** [review]
When I run the command without the `repo` argument, e.g. `python3 scripts/generate_changelog.py 1.79.0 origin`, I get this error:

Screen Shot 2022-10-20 at 11 40 14 AM

File: scripts/generate_changelog.py (hunk 1 above)
Comment (CodyFinn-Akamai) [reply scripts/generate_changelog.py:116]: What about allowing both? Then adding it if needed?

Comment (dwiley-akamai) [reply scripts/generate_changelog.py:116]: That works for me, just want to ensure a consistent format and make sure it isn't prone to errors

File: scripts/generate_changelog.py [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
+RELEASE='v' + sys.argv[1].strip('v')
+ORIGIN=sys.argv[2]
+try:
+    REPO=sys.argv[3]
+except Exception:
+    REPO=False
 
-START_INSERT=5
+DATE = datetime.datetime.now().strftime('%Y-%m-%d')
+START_INSERT=5 if REPO.lower() == 'manager' else 0
```
Comment (dwiley-akamai) [inline scripts/generate_changelog.py:14]:
```suggestion
START_INSERT=0 if (REPO and REPO.lower() != 'manager') else 5
```

File: scripts/generate_changelog.py [hunk 3]
```diff
@@ -49,8 +54,11 @@ def remove_pull_request_id(commit_message):
     regexp = "\(#\d+\)"
     return re.sub(regexp, '', commit_message).strip()
 
-def generateChangeLog(release, date, origin):
-    git_log_command = ["git", "log", "--no-merges", "--oneline", "--pretty='%s'", "{}/master...HEAD".format(origin)]
+def generateChangeLog(release, date, origin, repo=False):
```
Comment (dwiley-akamai) [inline scripts/generate_changelog.py:57]:
```suggestion
def generateChangeLog(release, date, origin, repo=''):
```

File: scripts/generate_changelog.py [hunk 4]
```diff
@@ -93,12 +101,19 @@ def generateChangeLog(release, date, origin):
 
     generateJQLQuery(jql_query)
 
-    read_change_log=open('CHANGELOG.md', 'r')
+    changelog_file = "CHANGELOG.md"
+
+    if (repo == 'manager'):
```
Comment (dwiley-akamai) [inline scripts/generate_changelog.py:106]:
```suggestion
    # If the repo isn't explicitly passed in, we default to Manager's CHANGELOG.
    if (repo == 'manager' or repo == '' or repo is False):
```

**Review by dwiley-akamai — COMMENTED** [review]
LGTM aside from the error I mentioned in the last comment when `repo` isn't passed as an argument. The suggestions below seemed to work for me but may need tweaking or could be less verbose

**Review by dwiley-akamai — APPROVED** [review]
Everything looks to be working as intended, nice work 🚀

**Review by bnussman-akamai — APPROVED** [review]

**Pull request merged**